packages/toolkit/src/helpers/error-handler.ts

```diff
@@ -25,82 +39,113 @@ export class ErrorHandler {
+                ?.reason || '',
+            ),
+        },
+        {
+          error: errors.ProviderInvalidInput,
+          getMessage: () => i18next.t('errors.provider-invalid-input'),
+        },
+        {
+          error: errors.ProviderResourceNotFound,
+          getMessage: () => i18next.t('errors.provider-resource-not-found'),
+        },
+        {
+          error: errors.ProviderResourceUnavailable,
+          getMessage: () => i18next.t('errors.provider-resource-unavailable'),
+        },
+        {
+          error: errors.ProviderTransactionRejected,
+          getMessage: () => i18next.t('errors.provider-transaction-rejected'),
+        },
+        {
+          error: errors.ProviderMethodNotSupported,
+          getMessage: () => i18next.t('errors.provider-method-not-supported'),
+        },
+        {
+          error: errors.ProviderLimitExceeded,
+          getMessage: () => i18next.t('errors.provider-limit-exceeded'),
+        },
+        {
+          error: errors.ProviderJsonRpcVersionNotSupported,
+          getErrorMessage: () =>
+            i18next.t('errors.provider-json-rpc-version-not-supported'),
+        },
+        {
+          error: errors.BadRequestError,
+          getErrorMessage: () =>
+            get(
+              { admin_not_found: i18next.t('errors.admin-not-found') },
+              get(error, 'code', '') ||
+                get(error, 'originalError.response.data.errors[0].code', ''),
               '',
-          )
-          break
-        case errors.ProviderInvalidInput:
-          errorMessage = i18next.t('errors.provider-invalid-input')
-          break
-        case errors.ProviderResourceNotFound:
-          errorMessage = i18next.t('errors.provider-resource-not-found')
-          break
-        case errors.ProviderResourceUnavailable:
-          errorMessage = i18next.t('errors.provider-resource-unavailable')
-          break
-        case errors.ProviderTransactionRejected:
-          errorMessage = i18next.t('errors.provider-transaction-rejected')
-          break
-        case errors.ProviderMethodNotSupported:
-          errorMessage = i18next.t('errors.provider-method-not-supported')
-          break
-        case errors.ProviderLimitExceeded:
-          errorMessage = i18next.t('errors.provider-limit-exceeded')
-          break
-        case errors.ProviderJsonRpcVersionNotSupported:
-          errorMessage = i18next.t(
-            'errors.provider-json-rpc-version-not-supported',
-          )
-          break
-        case errors.BadRequestError:
-          errorMessage = get(
-            { admin_not_found: i18next.t('errors.admin-not-found') },
-            get(error, 'code', '') ||
-              get(error, 'originalError.response.data.errors[0].code', ''),
-            '',
-          )
+            ),
+        },
+        ...this.#customErrors,
+      ]
+
+      for (const err of errorsList) {
+        if (error.constructor == err.error) {
```

Comment:
?

Reply:
why you check same variable in each iteration?
why you're checking by '==' and not by '==='?